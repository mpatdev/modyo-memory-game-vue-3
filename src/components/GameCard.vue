<script setup>
import { computed } from 'vue';

const props = defineProps({
  item: {
    type: Object,
    default: () => ({})
  },
  isFlipped: {
    type: Boolean,
    default: false
  },
  isFound: {
    type: Boolean,
    default: false
  }
})

const emit = defineEmits(['click'])

const altText = computed(() => {
  if(props.item.alt_text) {
    return props.item.alt_text
  }

  return props.item.title
})
</script>

<template>
  <div
    class="game-card overflow-hidden h-full border 
    border-black rounded cursor-pointer"
    :class="{
      'bg-zinc-200 flex items-center justify-center': !isFlipped
    }"
    @click="() => emit('click')"
  >
    <Transition name="flip">
      <div
        v-if="isFlipped"
        class="w-full h-full flex items-center p-2 justify-center"
      >
        <img
          :src="item.url"
          class="h-full w-auto"
          :title="altText"
          :alt="altText"
          :aria-label="altText"
          :role="none"
          :class="{
            'grayscale': isFound
          }"
        >
      </div>
      <div
        v-else
        class="w-full h-full flex items-center justify-center"
      >
        <div class="game-card__center w-20 h-20" />
      </div>
    </Transition>
  </div>
</template>

<style lang="scss" scoped>
.flip-enter-active {
  transition: all 0.8s ease;
}

.flip-leave-active {
  display: none;
}

.flip-enter-from, .flip-leave-from {
  transform: rotateY(180deg);
  opacity: 0;
}

.game-card {
  max-height: calc(25dvh - 1em);

  &__center {
    background-image: url('@/assets/images/card-background.png');
    background-repeat: no-repeat;
    background-size: cover;
  }
}
</style>