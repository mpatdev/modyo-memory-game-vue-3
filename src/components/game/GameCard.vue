<script setup>
import { computed } from 'vue'

const props = defineProps({
  item: {
    type: Object,
    default: () => ({})
  },
  isFlipped: {
    type: Boolean,
    default: false
  },
  isFound: {
    type: Boolean,
    default: false
  }
})

const emit = defineEmits(['click'])

const altText = computed(() => {
  if(props.item.alt_text) {
    return props.item.alt_text
  }

  return props.item.title
})
</script>

<template>
  <div
    class="game-card"
    :class="{
      'game-card--flipped': !isFlipped
    }"
    @click="() => emit('click')"
  >
    <Transition name="flip">
      <div
        v-if="isFlipped"
        class="w-full h-full flex items-center p-2 justify-center"
      >
        <img
          :src="item.url"
          class="h-full w-auto"
          :title="altText"
          :alt="altText"
          :aria-label="altText"
          :class="{
            'grayscale': isFound
          }"
        >
      </div>
      <div
        v-else
        class="w-full h-full flex items-center justify-center"
      >
        <img
          src="@/assets/images/question.png"
          alt="question sign"
          class="h-full"
        >
      </div>
    </Transition>
  </div>
</template>

<style lang="scss" scoped>
.flip-enter-active {
  transition: all 0.8s ease;
}

.flip-leave-active {
  display: none;
}

.flip-enter-from, .flip-leave-from {
  transform: rotateY(180deg);
  opacity: 0;
}

.game-card {
  @apply overflow-hidden h-full border-4 border-violet rounded cursor-pointer bg-secondary;
  max-height: calc(25dvh - 1em);

  &--flipped {
    @apply flex items-center justify-center;
  }
}
</style>