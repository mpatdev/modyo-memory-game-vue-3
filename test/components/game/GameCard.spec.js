import GameCard from '@/components/game/GameCard.vue'

import snapshotTest from '@test/snapshotTest'

snapshotTest(GameCard, 'GameCard')