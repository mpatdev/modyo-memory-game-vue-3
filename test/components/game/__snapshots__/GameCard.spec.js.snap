// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameCard > Check Snapshot 1`] = `
"<div data-v-284ef204=\\"\\" class=\\"game-card game-card--flipped\\">
  <transition-stub data-v-284ef204=\\"\\" name=\\"flip\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
    <div data-v-284ef204=\\"\\" class=\\"w-full h-full flex items-center justify-center\\"><img data-v-284ef204=\\"\\" src=\\"/src/assets/images/question.png\\" alt=\\"question sign\\" class=\\"h-full\\"></div>
  </transition-stub>
</div>"
`;
